### PR TITLE
Branch v1.3 diagrams

### LINKED ISSUE #109: Fix diagram issue for Developer Guide
1. Change image link in DG to point to the generated PlantUML diagram instead of the `puml` file
2. Add diagrams for the `list`, `reject` and command

### PR DESCRIPTION
**Pull request by Merrickneo**
## What
1. Edit broken links for the filepaths for activity diagrams
2. Refactor AddressBookStorage to HMHeroStorage
3. Add diagrams for the `list` and `reject` commands

Fix #109

### CODE REVIEW

File: docs/images/StorageClassDiagram.png
Comment (FredericChow00) [inline docs/images/StorageClassDiagram.png:1]: arrows are glitched out

**Review by FredericChow00 — CHANGES_REQUESTED** [review]
some minor changes

File: docs/images/RejectApplicantActivityDiagram.png
Comment (Jxleejiaxin) [inline docs/images/RejectApplicantActivityDiagram.png:1]: Missing check if applicant is already `REJECTED` status.

File: docs/diagrams/EditApplicantActivityDiagram.puml
Comment (Jxleejiaxin) [inline docs/diagrams/EditApplicantActivityDiagram.puml:1]: I am assuming this is a placeholder?

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 1]
```diff
@@ -51,6 +52,9 @@ public AddCommand parse(String args) throws ParseException {
         Status status = Status.APPLIED;
         ApplicationDateTime applicationDateTime = ParserUtil.parseApplied(argMultimap
                 .getValue(PREFIX_APPLIEDTIME).orElse(""));
+        if (applicationDateTime.getApplicationDateTime().isAfter(LocalDateTime.now())) {
+            throw new ParseException("Application Date cannot be in the future!");
```
Comment (Jxleejiaxin) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:56]: Need to edit `add` activity diagram and feature implementation in DG.

**Review by Jxleejiaxin — COMMENTED** [review]
Just some minor reminders. Thanks for refactoring everything!

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java (hunk 1 above)
Comment (Merrickneo) [reply src/main/java/seedu/address/logic/parser/AddCommandParser.java]: Ok done!

**Comment by Merrickneo** [comment]
I have to refactor some more classes for the `Storage` component! Will update here when i'm done.

**Review by FredericChow00 — APPROVED** [review]
LGTM!

**Pull request merged**